base/abstractarray.jl

```diff
@@ -246,8 +246,8 @@ end
 ## Traits for array types ##
 
 abstract LinearIndexing
-immutable LinearFast <: LinearIndexing end
-immutable LinearSlow <: LinearIndexing end
+struct LinearFast <: LinearIndexing end
+struct LinearSlow <: LinearIndexing end
```

Comment:
Yes, empty structs are fine in C++ and commonly used for unique tags with which traits can be associated, etc. In an amusing quirk of the C++ standard, `sizeof(boost::blank) != 0` :-(